N1_IntroToMonteCarlo.ipynb

```diff
@@ -0,0 +1,807 @@
+        "\n",
+        "The combination of Markov Chains with Monte Carlo sampling is extremely powerful, and is what we will explore in the next notebook.\n"
+      ],
+      "metadata": {
+        "id": "L0raUAP4yxy3"
+      }
+    },
+    {
+      "cell_type": "markdown",
+      "source": [
+        "#Exercises\n",
+        "\n",
+        "Complete the exercises below."
+      ],
+      "metadata": {
+        "id": "o6OP4uxq1ooP"
+      }
+    },
+    {
+      "cell_type": "markdown",
+      "source": [
+        "## Programming Exercise 1: Markov Chain\n",
+        "\n",
+        "Below is a partially completed program that simulates the Markov Chain model shown in the figure above.  \n",
+        "\n",
+        "a) Complete the program, and report the average number of steps it takes to change to state $A$ when starting in state $E$.  Include the uncertainty (i.e. [standard error](https://en.wikipedia.org/wiki/Standard_error)) in your reported number.  You should make additional code cells in your notebook to keep track of your data and present your results.\n",
+        "\n",
+        "\n",
+        "\n",
+        "---\n",
+        "Here is the transition probability table:\n",
+        "\n",
+        "State Change | Probability \n",
+        "--- | ---\n",
+        "EE | 0.3\n",
+        "EA | 0.7\n",
+        "AE | 0.4\n",
+        "AA | 0.6"
+      ],
+      "metadata": {
+        "id": "0nlkg2yqJngc"
+      }
+    },
+    {
+      "cell_type": "code",
+      "execution_count": 2,
+      "metadata": {
+        "id": "OZqVToXUx1qG"
+      },
+      "outputs": [],
+      "source": [
+        "# Here are some variables to get you started\n",
+        "\n",
+        "# Decide how many steps to make the simulation run\n",
+        "num_steps = 100\n",
+        "\n",
+        "# Define a starting state: \n",
+        "current_state = 'E' \n",
+        "\n",
+        "# Define an ending state:\n",
+        "end_state = 'A'"
+      ]
+    },
+    {
+      "cell_type": "code",
+      "source": [
+        "# Decide on a probability distribution function for the simulation \n",
+        "# (uncomment one of the probability lines in the function): \n",
+        "\n",
+        "def get_probability(): \n",
+        "  # This function returns a probability value when called\n",
+        "\n",
+        "  # --> This is a sample from a uniform distribution from numpy\n",
+        "  \n",
+        "  #prob = np.random.uniform(0, 1, 1)\n",
+        "\n",
+        "  # --> This is a sample from a normal distribution from numpy\n",
+        "  # --> It is centered at 0.5 to match the default uniform distribution\n",
+        "  \n",
+        "  #prob = np.random.normal(loc=0.5, size=1)\n",
+        "  \n",
+        "  return prob"
+      ],
+      "metadata": {
+        "id": "O4Wkbgt-dkCC"
+      },
+      "execution_count": 1,
+      "outputs": []
+    },
+    {
+      "cell_type": "code",
+      "source": [
+        "# Now we run the simulation\n",
```

Comment:
This cell works great, but I didn't immediately realize that you have to change the current state back to 'E' in the cell two cells above every time you rerun it. Maybe you could redefine current cell to 'E' in this cell?

Reply:
Thank you for your comment.  I have moved the cell with the `current_state` variable to be closer to the simulation cell.  I do not want to combine this variable with the simulation code because I want students to understand the difference between initializing the system and simulating the system.  The difficulty of executing two separate cells also motivates the development of the code function in the next exercise, where this variable is combined with the simulation code.